src/Features/Core/Portable/UnifiedSuggestions/UnifiedSuggestedActionsSource.cs

```diff
@@ -0,0 +1,711 @@
+            CodeFixCollection collection,
+            Workspace workspace)
+        {
+            var applicableFixes = collection.Fixes.WhereAsArray(f => IsApplicable(f.Action, workspace));
+            return applicableFixes.Length == 0
+                ? null
+                : applicableFixes.Length == collection.Fixes.Length
+                    ? collection
+                    : new CodeFixCollection(
+                        collection.Provider, collection.TextSpan, applicableFixes,
+                        collection.FixAllState, collection.SupportedScopes, collection.FirstDiagnostic);
+        }
+
+        /// <summary>
+        /// Arrange fixes into groups based on the issue (diagnostic being fixed) and prioritize these groups.
+        /// </summary>
+        private static ImmutableArray<UnifiedSuggestedActionSet> OrganizeFixes(
+            Workspace workspace,
+            ImmutableArray<CodeFixCollection> fixCollections,
+            bool includeSuppressionFixes)
+        {
+            var map = ImmutableDictionary.CreateBuilder<CodeFixGroupKey, IList<UnifiedSuggestedAction>>();
+            using var orderDisposer = ArrayBuilder<CodeFixGroupKey>.GetInstance(out var order);
+
+            // First group fixes by diagnostic and priority.
+            GroupFixes(workspace, fixCollections, map, order, includeSuppressionFixes);
+
+            // Then prioritize between the groups.
+            var prioritizedFixes = PrioritizeFixGroups(map.ToImmutable(), order.ToImmutable(), workspace);
+            return prioritizedFixes;
+        }
+
+        /// <summary>
+        /// Groups fixes by the diagnostic being addressed by each fix.
+        /// </summary>
+        private static void GroupFixes(
+           Workspace workspace,
+           ImmutableArray<CodeFixCollection> fixCollections,
+           IDictionary<CodeFixGroupKey, IList<UnifiedSuggestedAction>> map,
+           ArrayBuilder<CodeFixGroupKey> order,
+           bool includeSuppressionFixes)
+        {
+            foreach (var fixCollection in fixCollections)
+            {
+                ProcessFixCollection(
+                    workspace, map, order, includeSuppressionFixes, fixCollection);
+            }
+        }
+
+        private static void ProcessFixCollection(
+            Workspace workspace,
+            IDictionary<CodeFixGroupKey, IList<UnifiedSuggestedAction>> map,
+            ArrayBuilder<CodeFixGroupKey> order,
+            bool includeSuppressionFixes,
+            CodeFixCollection fixCollection)
+        {
+            var fixes = fixCollection.Fixes;
+            var fixCount = fixes.Length;
+
+            UnifiedSuggestedActionSet? getFixAllSuggestedActionSet(CodeAction codeAction)
+                => GetUnifiedFixAllSuggestedActionSet(
+                    codeAction, fixCount, fixCollection.FixAllState,
+                    fixCollection.SupportedScopes, fixCollection.FirstDiagnostic,
+                    workspace);
```

Comment:
would prefer this local function moved to bottom, with an explicit `return;` statement above it.

Reply:
(like how it is in the method below).